examples/main/main.spec.js

```diff
@@ -42,11 +47,225 @@ describe('examples/main', () => {
         'Another blog (another-blog)'
       ]
 
-      $('[data-test="contentarray-container"] > div').each(function(index, element) {
+      $('[data-test="contentarray-container"] > div').each(function (index, element) {
         expect($(this).children('span.data-title').html())
           .to.equal(expectedTitles[index]);
       })
     })
   })
-});
 
+  //another-blog example
+  describe('another-blog/example.html', () => {
+    let $;
+    before(() => {
+      $ = getSelector(path.join(__dirname, 'dist', 'another-blog', 'example.html'))
+    })
+
+    it('should render body text', () => {
+      expect($('body').html().trim())
+        .to.equal(String("another-blog"));
+    })
+  })
+
+  //another-blog index
+  describe('another-blog/index.html', () => {
+    let $;
+    before(() => {
+      $ = getSelector(path.join(__dirname, 'dist', 'another-blog', 'index.html'))
+    })
+
+    it('should render header text', () => {
+      expect($('[id="another-blog"]').html())
+        .to.equal('Another blog');
+    })
+
+    it('should render all blogs into blog container', () => {
+      let blogs = [
+        '..',
+        'another-blog',
+        'another-blog',
+      ]
+
+      $('#blog-container span').each(function (index, element) {
+        expect($(this).html()).to.equal(blogs[index]);
+      });
+    })
+
+    it('should render first para text', () => {
+      let dateToCheck = new Date();
+      dateToCheck.setFullYear(2020, 4, 9);
+      dateToCheck.setHours(0);
+      dateToCheck.setMinutes(0);
+      dateToCheck.setSeconds(0);
+      expect($('body main section p').first().html())
+        .to.equal(dateToCheck.toString());
+    })
+
+    it('should render last para text', () => {
+      expect($('body main section p').last().html())
+        .to.equal(String("Amazing blog right"));
+    })
+
+  });
+
+  //deep 
+  describe('deep/index.html', () => {
+    let $;
+    before(() => {
+      $ = getSelector(path.join(__dirname, 'dist', 'deep', 'index.html'))
+    })
+
+    it('should render html body text', () => {
+      expect($('body').html())
+        .to.equal(String("ok .."));
+    })
+  })
+
+  //more deep 
+  describe('deep/moredeep/index.html', () => {
+    let $;
+    before(() => {
+      $ = getSelector(path.join(__dirname, 'dist', 'deep', 'moredeep', 'index.html'))
+    })
+
+    it('should render html body text', () => {
+      expect($('body').html())
+        .to.equal(String(`..${path.sep}..`));
+    })
+  })
+
+
+  //my first blog example  
+  describe('my-first-blog/example.html', () => {
+    let $;
+    before(() => {
+      $ = getSelector(path.join(__dirname, 'dist', 'my-first-blog', 'example.html'))
+    })
+
+    it('should render html body text', () => {
+      expect($('body').text().trim())
+        .to.equal(String("my-first-blog"));
```

Comment:
Also, I don't think so `String()` is required anywhere here. Most of them are already strings. String parses other types to string so if you have a number and you wrap it in a String, it becomes a number in a String `3` becomes `"3"`. For the values that are already String (inside `""`), this may end up giving unexpected results

Reply:
Ohh,let me try..